gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -252,40 +249,41 @@ public void launchJob(Properties jobProps, JobListener jobListener)
       // Start the job and wait for it to finish
       runJob(jobName, jobProps, jobState, workUnits.get());
 
-      // Check and set final job state upon job completion
+      // Check and set final job jobPropsState upon job completion
       if (jobState.getState() == JobState.RunningState.CANCELLED) {
         LOG.info(String.format("Job %s has been cancelled", jobId));
         return;
       }
-      setFinalJobState(jobState);
 
+      JobCommitPolicy commitPolicy = JobCommitPolicy.forName(
+          jobState.getProp(ConfigurationKeys.JOB_COMMIT_POLICY_KEY, ConfigurationKeys.DEFAULT_JOB_COMMIT_POLICY));
```

Comment:
In that case, I suggest that we have a helper method JobPolicy.getCommitPolicy(). Copying/pasting the above line wherever policy is needed is error-prone.